### ISSUE TITLE
Fetch feedback fails when using multiple courses in assignment list

### ISSUE THREAD

**Issue by QuantumEntangledAndy** [jupyter/nbgrader#1178]
### Notes

I have set it up for multiple courses per students with multiple teachers. This is actually all working well except for fetch feedback.

I describe a patch to allow "fetch_feedback" to work but would like comments before making a pull request, as I expect it can be done better.

### Operating system

- Server on Ubuntu 18.04 LTS

- Client tried on osx and windows

### `nbgrader --version`

```
Python version 3.6.8 (default, Jan 14 2019, 11:02:34)
[GCC 8.0.1 20180414 (experimental) [trunk revision 259383]]
nbgrader version 0.6.0.dev
```

### `jupyterhub --version` (if used with JupyterHub)

```
1.0.0
```

### `jupyter notebook --version` 

```
6.0.0
```

### Expected behavior

To be able to collect feedback from the assignment list, (as a student)

### Actual behavior

Api request fails. Checking the response it reports

```
No course id specified. Re-run with --course flag.
```

### Discussion

This seems to be caused by the courseid, which is passed in the header, not being used in the `class FetchFeedbackHandler` of `nbgrader/server_extensions/formgrader/apihandlers.py`

### Patch

I have patched this to work on my server by editing

```python
class FetchFeedbackHandler(BaseApiHandler):
    @web.authenticated
    @check_xsrf
    @check_notebook_dir
    def post(self, assignment_id, student_id):
        ret_dict = {}
        ret_dict.update(self.api.fetch_feedback(assignment_id, student_id))
        self.write(json.dumps(ret_dict))
```

to 

```python
class FetchFeedbackHandler(BaseApiHandler):
    @web.authenticated
    @check_xsrf
    @check_notebook_dir
    def post(self, assignment_id, student_id):
        course_id = self.get_argument('course_id')
        self.api.coursedir.course_id = course_id
        ret_dict = {}
        ret_dict.update(self.api.fetch_feedback(assignment_id, student_id))
        self.write(json.dumps(ret_dict))
```

So that it uses the 'course_id' data in the header. I'm not sure if this should be made into a pull request as I expect it doesn't cover all cases. Perhaps it should only be done like this when using multiple courses in the assignment list.

Also is it intentional that the assignment list make calls to the formgrader server extension? I was under the impression that students only required access to the assignment list extension but perhaps I was mistaken? I had to enable the formgrader server extension for all the students as well as apply the above patch.

**Comment by jhamrick** [comment]
Thanks for catching this! I think the course id does indeed need to be specified here. Though I would probably do this by modifying the signature of `self.api.fetch_feedback` to accept the course id and then set it as a temporary attribute, rather than setting it directly on `self.api.coursedir.course_id`. If you want to make a PR for this that would be great!

> Also is it intentional that the assignment list make calls to the formgrader server extension? I was under the impression that students only required access to the assignment list extension but perhaps I was mistaken? I had to enable the formgrader server extension for all the students as well as apply the above patch.

Ah very good catch! You're right, this handler shouldn't be in the formgrader; it should be in the assignment list. I've opened a separate issue to track this: #1181

**Comment by jhamrick** [comment]
My guess is this was fixed by #1184, but could you verify it's working for you now?

**Comment by QuantumEntangledAndy** [comment]
Yes this PR #1184 does indeed fix the issue.

But as mentioned in the PR I am also seeing double the number of current assignments after fetching. Will follow the appropriate issue #1188 for that.

Closing this.
